azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -320,6 +320,38 @@ public synchronized Project purgeProject(Project project, User deleter)
         return project;
     }
 
+  /**
+   * Permanently delete project files with old versions (file with highest version not purged)
+   * and properties data for all except the current version of a project and log event
+   * in project_events table.
+   *
+   * @param project
+   * @param deleter
+   * @return
+   * @throws ProjectManagerException
+   */
+  public synchronized Project purgeActiveProject(Project project, User deleter)
+      throws ProjectManagerException {
+
+    /*
+     * Check if max version matches. If the project version is equal to max of
+     * version in table project_versions, we do the purging; Otherwise, exception
+     * is thrown.
+     */
+    if (projectLoader.getLatestProjectVersion(project) == project.getVersion()) {
```

Comment:
Maybe explain the reason for this check? This is there because of the assumptions the cleanOlderProjectVersion made, right?